crates/ra_ssr/src/resolving.rs

```diff
@@ -157,9 +170,15 @@ impl<'db> ResolutionScope<'db> {
         ResolutionScope {
             scope,
             hygiene: hir::Hygiene::new(sema.db, resolve_context.file_id.into()),
+            node,
         }
     }
 
+    /// Returns the function in which SSR was invoked, if any.
+    pub(crate) fn current_function(&self) -> Option<SyntaxNode> {
+        self.node.ancestors().find(|node| node.kind() == SyntaxKind::FN).map(|node| node.clone())
```

Comment:
This seems like it might be invoked a lot, maybe it should be precomputed and stored in a field? Other than that this looks good to me.

Reply:
It's invoked once for each rule that contains `self`. Rules containing `self` are probably moderately uncommon, so most of the time it won't be invoked at all and when it is invoked, it'll probably only be once a lot of the time, since when self is present, there's a good chance that the rule will only parse as an expression. That said, I don't object to storing it in a field

Reply:
Ah, that seems fine then.